byzcoin/tx_pipeline.go

```diff
@@ -483,12 +460,41 @@ func (p *txPipeline) processTxs(initialState *txProcessorState) {
 						}
```

Comment:
While we're at it we could refactor this go routine to:

```
func(state *txProcessorState) {
	p.wg.Add(1)
	defer p.wg.Done()
	if state != nil {
		// NOTE: ProposeBlock might block for a long time,
		// but there's nothing we can do about it at the moment
		// other than waiting for the timeout.
		err := p.processor.ProposeBlock(state)
		if err != nil {
			log.Error("failed to propose block:", err)
			proposalResult <- err
			return
		}
	}
	proposalResult <- nil
}(inState)
```

(I can do it in another PR if you want)